app/models/report.rb

```diff
@@ -33,4 +34,16 @@ def next
           .order(:reported_on)
           .first
   end
+
+  private
+
+    def learning_times_correct?
+      learning_times.all? { |learning_time| learning_time.diff > 0 }
+    end
+
+    def learning_times_finished_at_be_greater_than_started_at
+      if !wip? && !learning_times_correct?
```

Comment:
```suggestion
      if !wip? && !learning_time_zero?
```

```suggestion
      if !wip? && learning_time > 0
```

💭 みたいな方があとから読みやすいかも

validateというチェックメソッドの中のメソッドでcorrect(あってるか)？というまたチェックOK?みたいな名前出されるよりもチェックメソッドのチェック内容がわかるメソッド（学習時間が0ですか？）が中にある方がわかりやすそうです。

Reply:
↑賛成ー！